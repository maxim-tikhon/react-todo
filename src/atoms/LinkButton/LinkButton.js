import styles from "./LinkButton.module.scss";

const LinkButton = ({ children, isActive, ...props }) => {
  const className = isActive
    ? `${styles.linkButton} ${styles.active}`
    : styles.linkButton;

  return (
    <button className={className} {...props}>
      {children}
    </button>
  );
}

export default LinkButton;
