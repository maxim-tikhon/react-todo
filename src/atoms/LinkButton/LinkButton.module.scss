.linkButton {
	background: none;
	border: none;
	color: grey;
	cursor: pointer;
	font-size: 14px;

	&.active {
		color: rgb(48, 117, 229);
	}

	&:hover {
		color: black;
	}
}
