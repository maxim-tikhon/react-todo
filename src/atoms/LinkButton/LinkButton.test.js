import renderer from 'react-test-renderer';
import LinkButton from './LinkButton';

describe('LinkButton', () => {
	it('renders correctly', () => {
		const output = renderer.create(<LinkButton isActive>Test Bustton</LinkButton>).toJSON();
		expect(output).toMatchSnapshot();
	});
});
