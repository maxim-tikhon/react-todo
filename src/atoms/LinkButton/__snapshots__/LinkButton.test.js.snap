// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders correctly 1`] = `
<button
  className="linkButton active"
  type="button"
>
  Test Bustton
</button>
`;
